src/JsonNormalizer.php

```diff
@@ -33,15 +33,15 @@ public static function asNormalizedJson(array $structure) : string
      * @param mixed[] $structure
      *     The array of JSON to sort, passed by reference.
      *
-     * @throws \Exception
+     * @throws \RuntimeException
      *     Thrown if sorting the array fails.
      *
      * @return void
      */
     private static function rKeySort(array &$structure) : void
     {
         if (!ksort($structure, SORT_STRING)) {
-            throw new \Exception("Failure sorting keys. Canonicalization is not possible.");
+            throw new \RuntimeException("Failure sorting keys. Canonicalization is not possible.");
```

Comment:
So this lightweight JSON canonicalization that we're implementing is a source of problems (exactly related to what David and Ted were debugging today), and this seems like a way to insert a vulnerability. "Feed it garbage JSON that causes a sort error, and then halt their updater." I feel we might want to step up a level at this whole API is a separate issue.